enclave-runtime/src/rpc/author/mod.rs

```diff
@@ -100,21 +98,20 @@ pub trait OnBlockCreated {
 }
 
 /// Authoring API
-pub struct Author<'a, P>
+pub struct Author<TopPoolGetter>
 where
-	P: TrustedOperationPool + Sync + Send + 'static,
+	TopPoolGetter: GetTopPool + Sync + Send + 'static,
 {
-	/// Trusted Operation pool
-	pool: Arc<&'a P>,
+	top_pool_getter: Arc<TopPoolGetter>,
```

Comment:
The author now has a `GetTopPool` instance, instead of a reference to a pool (which originated from a mutex and has very limited scope).